cloud/google/clusteractuator.go

```diff
@@ -94,19 +108,27 @@ func (gce *GCEClusterClient) Reconcile(cluster *clusterv1.Cluster) error {
 	if err != nil {
 		glog.Warningf("Error creating firewall rule for core api server traffic: %v", err)
 	}
+
+	// If any resources had to be created for this cluster, emit a created event.
```

Comment:
I don't follow the logic here -- if we created a firewall then we tell the user (via an event) that we created a cluster? What if someone deleted a firewall and we reconcile it back into existence -- then we emit an event that a cluster was created (again)?

Reply:
I don't want the created event to fire every time the reconcile method is called, so I need to check if something was actually created. But I also want it to emit the event if the reconcile method failed halfway through in a previous run.

Eventually this will create more resources in addition to the firewall rules. For example, let's pretend this also creates service accounts after the firewall rules, which it will eventually. If the firewall rules created successfully, but the service account failed to create because of some misconfiguration in the cluster object, a Warning event will be emitted for the error. Then if the operator fixes the misconfiguration, the firewall rules will not be created again on the next run, but the service accounts now can be. In that case, I want the Created event to be emitted on the second run. 

Does that make sense? I'm open to suggestions, this seems to be confusing.

Reply:
This sounds ok for now, but I think we may need to think about a better way to represent the status or condition of a cluster object since it's multifaceted.